keras_cv/layers/object_detection/rpn_label_encoder.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Mapping
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+from keras_cv.bounding_box import iou
+from keras_cv.ops import box_matcher
+from keras_cv.ops import sampling
+from keras_cv.ops import target_gather
+
+
+class _RpnLabelEncoder(tf.keras.layers.Layer):
+    """Transforms the raw labels into training targets for region proposal network (RPN).
+
+    # TODO(tanzhenyu): consider unifying with _ROISampler.
+    This is different from _ROISampler for a couple of reasons:
+    1) This deals with unbatched input, dict of anchors and potentially ragged labels
+    2) This deals with gt boxes, while _ROISampler deals with padded gt
```

Comment:
For public docstring, we should change gt to ground truth.

Reply:
Done.